account_payment_order/models/account_move_line.py

```diff
@@ -95,7 +103,7 @@ def _get_communication(self):
                     self.move_id.move_type in ("in_invoice", "in_refund")
                     and self.move_id.ref
                 ):
-                    communication = self.move_id.payment_reference or self.move_id.ref
+                    communication = self.move_id.ref
```

Comment:
Why this is removed and why so many places to take the same thing? Shouldn't be one of there should be a reason to be some of them, and you need to take each use case apart.